### PR TITLE
Upgrade to the v0.12 release of the Terraform SDK

### PR DESCRIPTION
**Pull request by pasali**

### CODE REVIEW

**Comment by matpimenta** [comment]
Hi @pasali. Thanks for the PR. That's very helpful.
We will need to perform some tests to verify the impact of these changes (in terms of backwards compatibility, etc) before we merge it.
We'll come back to you as soon as we can.

File: go.sum [hunk 1]
```diff
@@ -88,121 +165,233 @@ github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1 h1:EGx4pi6eqNxGa
...
 github.com/hashicorp/memberlist v0.0.0-20170208211506-23ad4b7d7b38/go.mod h1:ncdBp14cuox2iFOq3kDiquKU6fqsTBc3W6JvZwjxxsE=
+github.com/hashicorp/memberlist v0.1.0/go.mod h1:ncdBp14cuox2iFOq3kDiquKU6fqsTBc3W6JvZwjxxsE=
+github.com/hashicorp/serf v0.0.0-20160124182025-e4ec8cc423bb/go.mod h1:h/Ru6tmZazX7WO/GDmwdpS975F019L4t5ng5IgwbNrE=
 github.com/hashicorp/serf v0.8.2-0.20171022020050-c20a0b1b1ea9 h1:SYvpFFTluyu7KQoR1vFbk72jMMeXa8TedEo9VihOiI4=
 github.com/hashicorp/serf v0.8.2-0.20171022020050-c20a0b1b1ea9/go.mod h1:h/Ru6tmZazX7WO/GDmwdpS975F019L4t5ng5IgwbNrE=
 github.com/hashicorp/terraform v0.11.12-beta1.0.20190214175014-182daa619826 h1:Mm3O+waTD8/+yOVyO53M8ALXK5R5rDE8J3RjXni77rg=
 github.com/hashicorp/terraform v0.11.12-beta1.0.20190214175014-182daa619826/go.mod h1:8tUWsL56EhJHEy45QX9SROGT3+GSU06FVkh6GsP5uSY=
+github.com/hashicorp/terraform v0.12.0-rc1 h1:lxQadsdLSFQD/lfY1u/cfqHUck5lq5HQ4GOCzQkYoEU=
+github.com/hashicorp/terraform v0.12.0-rc1/go.mod h1:mLbSyFcATc332aCwDeX9cziNg/dUb6ZJzRVd1KgrDOk=
+github.com/hashicorp/terraform v0.12.0 h1:B5lTDWSdEbbjxrTZQ8GdsDlH7FnDrIvF42GykbhQ6tA=
```
Comment (mikecook) [inline go.sum:178]: This line is somehow outdated. Remove it and rerun `go mod tidy` && `go mod vendor`.

Before line removal:
```
$ GO111MODULE=on go mod tidy
go: finding github.com/sethvargo/go-fastly v0.0.0-20181208165200-c9e8e74a07f0
go: finding github.com/google/jsonapi v0.0.0-20180313013858-2dcc18f43696
verifying github.com/hashicorp/terraform@v0.12.0: checksum mismatch
	downloaded: h1:It2vmod2dBMB4+r+aUW2Afx0HlftyUwzNsNH3I2vrJ8=
	go.sum:     h1:B5lTDWSdEbbjxrTZQ8GdsDlH7FnDrIvF42GykbhQ6tA=
```
After line removal and rerunning tidy and vendor, these lines are changed (in addition to many others):
```
-github.com/hashicorp/terraform v0.12.0 h1:B5lTDWSdEbbjxrTZQ8GdsDlH7FnDrIvF42GykbhQ6tA=
+github.com/hashicorp/terraform v0.12.0 h1:It2vmod2dBMB4+r+aUW2Afx0HlftyUwzNsNH3I2vrJ8=
```

File: go.sum [hunk 2]
```diff
@@ -88,121 +165,233 @@ github.com/gopherjs/gopherjs v0.0.0-20181017120253-0766667cb4d1 h1:EGx4pi6eqNxGa
...
 github.com/hashicorp/go-checkpoint v0.0.0-20171009173528-1545e56e46de h1:XDCSythtg8aWSRSO29uwhgh7b127fWr+m5SemqjSUL8=
 github.com/hashicorp/go-checkpoint v0.0.0-20171009173528-1545e56e46de/go.mod h1:xIwEieBHERyEvaeKF/TcHh1Hu+lxPM+n2vT1+g9I4m4=
+github.com/hashicorp/go-checkpoint v0.5.0/go.mod h1:7nfLNL10NsxqO4iWuW6tWW0HjZuDrwkBuEQsVcpCOgg=
 github.com/hashicorp/go-cleanhttp v0.5.0 h1:wvCrVc9TjDls6+YGAF2hAifE1E5U1+b4tH6KdvN3Gig=
 github.com/hashicorp/go-cleanhttp v0.5.0/go.mod h1:JpRdi6/HCYpAwUzNwuwqhbovhLtngrth3wmdIIUrZ80=
 github.com/hashicorp/go-getter v0.0.0-20180327010114-90bb99a48d86 h1:hLYM35twiyKH44g36g+GFYODcrZQetEAY4+zrJtGea0=
 github.com/hashicorp/go-getter v0.0.0-20180327010114-90bb99a48d86/go.mod h1:6rdJFnhkXnzGOJbvkrdv4t9nLwKcVA+tmbQeUlkIzrU=
+github.com/hashicorp/go-getter v1.1.0 h1:fb6yisdpoNLiQZsuZqafiuaSSXFjxByBm/YS3k8MXjo=
+github.com/hashicorp/go-getter v1.1.0/go.mod h1:q+PoBhh16brIKwJS9kt18jEtXHTg2EGkmrA9P7HVS+U=
+github.com/hashicorp/go-getter v1.3.0 h1:zVkuohrXQAfI32em/DcGmurxxVwrnJk+m5/KybEPLSw=
```
Comment (mikecook) [inline go.sum:141]: This line is somehow outdated. Remove it and rerun `go mod tidy` && `go mod vendor`.

Before line removal:
```
$ GO111MODULE=on go mod tidy
go: downloading github.com/sethvargo/go-fastly v0.0.0-20181208165200-c9e8e74a07f0
verifying github.com/hashicorp/go-getter@v1.3.0: checksum mismatch
	downloaded: h1:pFMSFlI9l5NaeuzkpE3L7BYk9qQ9juTAgXW/H0cqxcU=
	go.sum:     h1:zVkuohrXQAfI32em/DcGmurxxVwrnJk+m5/KybEPLSw=
```

After line removal and rerunning tidy and vendor, these lines are changed (in addition to many others):
```
-github.com/hashicorp/go-getter v1.3.0 h1:zVkuohrXQAfI32em/DcGmurxxVwrnJk+m5/KybEPLSw=
+github.com/hashicorp/go-getter v1.3.0 h1:pFMSFlI9l5NaeuzkpE3L7BYk9qQ9juTAgXW/H0cqxcU=
```

Comment (pasali) [reply go.sum:141]: i did what you suggest but the line remains unchanged. i am not familiar with go dependency system. Maybe i am missing something

**Comment by KenticoMartinS** [comment]
Hi @matpimenta, 
I wanna ask, how does it look with this PR? Will it be merged in the near future? We are desperately waiting for 0.12 support so we can perform the upgrade.

**Comment by trentrosenbaum** [comment]
Hi @KenticoMartinS

I just wanted to get back to you about the upgrade to 0.12.  Right now this is a priority for the provider and it is actively being worked on.  Currently, several acceptance tests are being investigated to confirm if there is a wider problem with the upgrade.

**Comment by jamessthompson** [comment]
Any idea of an ETA for this?  Fastly support is our one blocker to migrating to 0.12

**Comment by trentrosenbaum** [comment]
Hi @jamessthompson, right now we are in our final stages of review and testing to confirm the update.  As the master branch has moved forward we want to get those pieces out before the update.  The following draft PR https://github.com/terraform-providers/terraform-provider-fastly/pull/173 will be the focus after a rebase.

**Comment by jamessthompson** [comment]
Thanks for the update.

**Comment by philippschulte** [comment]
I'll close this out in favor of #173.

**Pull request closed without merging**